### PR TITLE
Platformer game

### PR DESCRIPTION
**Pull request by Roodiger**
A test platformer game

### CODE REVIEW

**Comment by wolffles** [comment]
Awesome I appreciate the contribution. 

In the Interest of time, could you do me a favor and supply me  the `example2.js` script that's being required in index.html? I want to test it locally before I merge.

I think It would be a good idea for future contributors if we add a ReadMe file to the brickbreak and platformer folders with sample code for a basic working version. ( if I get a working version of your game I'll add the READMEs myself)

**Comment by Roodiger** [comment]
The example2.js file should be included there in the pull. You should be able to grab it there. 

Yeah that's a good call. I'll work on some sort of ReadMe if you don't get to it first

**Comment by wolffles** [comment]
> The example2.js file should be included there in the pull. You should be able to grab it there.

I think it was ignored by the .gitignore file. I'm looking at the diff files and there isn't one

**Comment by Roodiger** [comment]
@wolffles I'll look into it in a minute

**Comment by Roodiger** [comment]
It should be there now? I made some slight changes. Sorry I'm getting used to GitHub. Let me know if you need anything.

File: node_modules/game-iverse/brickbreak/brickbreakclasses.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    };
+}
+
+// module.exports = {
+//     World: World,
+//     Hud:Hud,
+//     Brick:Brick,
+//     Ball:Ball,
+//     Paddle:Paddle
+// }
```
Comment (wolffles) [inline node_modules/game-iverse/brickbreak/brickbreakclasses.js:189]: @Roodiger why does your version have a `node modules/` folder with our package inside of it? This whole repo is the npm package and should not have our repo inside of it. Please remove it. 

The idea is to supply building blocks for others to use. I should have clarified that the html and js file you were supplying was for testing purposes only, and would be removed and added to a read me file upon merging. If you want to test your package create a test repo to `npm link` your two repos together.

File: node_modules/game-iverse/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const brickBreakClasses = require('./brickbreak/brickbreakclasses');
+const platformerClasses = require('./platformer/platformerclasses');
+
+module.exports.brickbreak = brickBreakClasses;
+module.exports.platformer = platformerClasses;
```
Comment (wolffles) [inline node_modules/game-iverse/index.js:5]: `node_module/` remove

File: node_modules/game-iverse/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/wolffles/game-iverse.git"
+  },
+  "scripts": {
+    "build": "webpack --watch",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "version": "1.2.1"
+}
```
Comment (wolffles) [inline node_modules/game-iverse/package.json:55]: `node_module/` remove

File: node_modules/game-iverse/platformer/levels.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+........#=..................=................=#...#.....................###...................................
+........#######################################...#+++++++++++++++++++++###+++++++++++++++++++++++++++++++++++
+..................................................############################################################
+..............................................................................................................
+`];
+
+if (typeof module != "undefined" && module.exports && (typeof window == "undefined" || window.exports != exports))
+  module.exports = GAME_LEVELS;
+if (typeof global != "undefined" && !global.GAME_LEVELS)
+  global.GAME_LEVELS = GAME_LEVELS;
```
Comment (wolffles) [inline node_modules/game-iverse/platformer/levels.js:178]: You should move this to the `platformer/READEME.md` and your test repository, I would like to try it out myself.
After I publish the new version we'll make a webpage to display our games. its just an NPM package shouldn't have a html doc to run. It should just be modules to be exported and imported. 

However in the npm package we shouldn't have a full game.

File: node_modules/game-iverse/platformer/platformer.html [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  runGame(GAME_LEVELS, DOMDisplay);
+
+
+  </script>
+</br>
+</body>
+
+
+</body>
```
Comment (wolffles) [inline node_modules/game-iverse/platformer/platformer.html:41]: Should place the `platformer/` folder in the root of the repo not in this `node_module/`

File: node_modules/game-iverse/platformer/platformer.js [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+}
+
+async function runGame(plans, Display) {
+  for (let level = 0; level < plans.length;) {
+    let status = await runLevel(new Level(plans[level]),
+                                Display);
+    if (status == "won") level++;
+  }
+  console.log("You've won!");
+}
```
Comment (wolffles) [inline node_modules/game-iverse/platformer/platformer.js:368]: I'm sorry you've done so much work, this is great just should be on our website advertising the package not in the package. We'll work on the site after.

File: package-lock.json [hunk 7]
```diff
@@ -1,5 +1,11 @@
...
-  "lockfileVersion": 1
+  "requires": true,
+  "lockfileVersion": 1,
+  "dependencies": {
+    "game-iverse": {
+      "version": "1.2.1",
+      "resolved": "https://registry.npmjs.org/game-iverse/-/game-iverse-1.2.1.tgz",
+      "integrity": "sha512-WoQ2YlQh7ncup9ILzadNGHA0MhWwIR2fSThhzCFtxEJb7dNJ2UWQBEkIsJRP0XoXwOgEWiVzINABFUiTyIO3qg=="
+    }
+  }
```
Comment (wolffles) [inline package-lock.json:10]: please don't modify the package-lock file.

File: node_modules/game-iverse/platformer/style.css [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+.won .player {
+box-shadow: -4px -7px 8px white, 4px -7px 8px white;
+}
+
+.game {
+overflow: hidden;
+max-width: 600px;
+max-height: 450px;
+position: relative;
+}
```
Comment (wolffles) [inline node_modules/game-iverse/platformer/style.css:24]: belongs on the website.

**Review by wolffles — CHANGES_REQUESTED** [review]
There seems to be a misunderstanding of what this project is. I really like the enthusiasm in creating all  the code. this project should just be modules to export, the reason I asked for the JS file was to test to see if your code worked on my test repository. On my local system I have the game-iverse repo, and a separate test-repo which I `npm link` to the game-iverse repo and test if it works like an npm package.
the examples are just guidelines for a user to understand how it works.  I'll have to make a contribution file to explain this for future contributors

**Pull request closed without merging**